flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,434 @@
+        if key in self.ALIASES:
+            return self.ALIASES[key]
+        return key
+
+    def _opts(self, key):
+        """Gets the options for the specified key"""
+        return self.PARAMS[self._key(key)]
+
+    def _getval(self, key, val):
+        """Gets the value for the specified key based on a config option"""
+        opts = self._opts(key)
+        if isinstance(opts, dict):
+            # Translate the input value to the API value
+            # The str cast converts bools to 'True'/'False' for use as keys
+            # This allows for options that have True/False/Other values
+            return opts[str(val)]
+        elif isinstance(val, list):
+            # Fix yaml parser making a list out of a string
+            return ",".join(val)
+        return val
+
+    def params_from_config(self, config):
+        """Filter params and map config values -> api values"""
+        ret = {}
+        for k, v in config.items():
+            key = self._key(k)
+            if key in self.PARAMS:
+                ret[key] = self._getval(k, v)
+        return ret
+
+    def setup(self, task, config):
+        """Set up a session and log in"""
+        self._session = Session()
+        base_url = config.get('base_url', "").rstrip("/")
+        if base_url:
+            if self.base_url and self.base_url != base_url:
+                log.warning("Using plugin designed for %s on %s - "
+                            "things may break", self.base_url, base_url)
+            self.base_url = base_url
+
+        if not self.base_url:
+            raise PluginError("No 'base_url' configured")
+
+        # The consistent request limiting rule seems to be:
+        # "Refrain from making more than five (5) requests every ten (10) seconds"
+        self._session.add_domain_limiter(TokenBucketLimiter(self.base_url, 2, '2 seconds'))
+
+        # Custom user agent
+        user_agent = config.get('user_agent', "Flexget [Gazelle plugin]")
+        if user_agent:
+            self._session.headers.update({"User-Agent": user_agent})
+        self.username = config['username']
+        self.password = config['password']
+
+        # Login
+        self.authenticate()
+
+        # Logged in successfully, it's ok if nothing matches now
+        task.no_entries_ok = True
+
+
+    def authenticate(self, force=False):
+        """
+        Log in and store auth data from the server
+        Adapted from https://github.com/isaaczafuta/whatapi
+        """
+
+        # clean slate before creating/restoring cookies
+        self._session.cookies.clear()
+
+        if not force:
+            with FlexgetSession() as session:
+                db_session = session.query(GazelleSession).filter(
+                    GazelleSession.base_url == self.base_url,
+                    GazelleSession.username == self.username
+                ).one_or_none()
+                if (db_session and db_session.expires and
```

Comment:
parentheses are redundant here

Reply:
I'm assuming this is more about the wrapping and less about the method of doing it? I was keeping to 80 chars out of habit but have just noticed that 120 is what's configured in setup.py - I'll unwrap the line since it's only 100 unwrapped.